dagshub/data_engine/model/datasource.py

```diff
@@ -543,6 +551,13 @@ def _upload_metadata(self, metadata_entries: List[DatapointMetadataUpdateEntry])
         # Update the status from dagshub, so we get back the new metadata columns
         self.source.get_from_dagshub()
 
+    def delete_metadata(self, path, id, key):
+        metadata_entries = [DatapointDeleteMetadataEntry(datapointId=id, url=path, key=key)]
+        self.source.client.delete_metadata(self, metadata_entries)
+
+        # Update the status from dagshub, so we get back the new metadata columns
+        self.source.get_from_dagshub()
```

Comment:
I don't see anything from frontend/backend that would make it so the whole metadata field is disappearing, unless I'm missing something.
If the metadata field can't disappear from the whole datasource after deletion, then there's no need to `get_from_dagshub()`